x-pack/plugins/remote_clusters/server/routes/api/delete_route.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get } from 'lodash';
+import { schema } from '@kbn/config-schema';
+import { i18n } from '@kbn/i18n';
+import { RequestHandler } from 'src/core/server';
+
+import { RouteDependencies } from '../../types';
+import { serializeCluster } from '../../../common/lib';
+import { API_BASE_PATH } from '../../../common/constants';
+import { doesClusterExist } from '../../lib/does_cluster_exist';
+import { licensePreRoutingFactory } from '../../lib/license_pre_routing_factory';
+import { isEsError } from '../../lib/is_es_error';
+
+export const register = (deps: RouteDependencies): void => {
+  const deleteHandler: RequestHandler<any, any, any> = async (ctx, request, response) => {
+    try {
+      const callAsCurrentUser = ctx.core.elasticsearch.dataClient.callAsCurrentUser;
+
+      const { nameOrNames } = request.params;
+      const names = nameOrNames.split(',');
+
+      const itemsDeleted: any[] = [];
+      const errors: any[] = [];
+
+      // Validator that returns an error if the remote cluster does not exist.
+      const validateClusterDoesExist = async (name: string) => {
+        try {
+          const existingCluster = await doesClusterExist(callAsCurrentUser, name);
+          if (!existingCluster) {
+            return response.customError({
+              statusCode: 404,
+              body: {
+                message: i18n.translate(
+                  'xpack.remoteClusters.deleteRemoteCluster.noRemoteClusterErrorMessage',
+                  {
+                    defaultMessage: 'There is no remote cluster with that name.',
+                  }
+                ),
+              },
+            });
+          }
+        } catch (error) {
+          return response.customError({ statusCode: 400, body: error });
+        }
+      };
+
+      // Send the request to delete the cluster and return an error if it could not be deleted.
+      const sendRequestToDeleteCluster = async (name: string) => {
+        try {
+          const body = serializeCluster({ name });
+          const updateClusterResponse = await callAsCurrentUser('cluster.putSettings', { body });
+          const acknowledged = get(updateClusterResponse, 'acknowledged');
+          const cluster = get(updateClusterResponse, `persistent.cluster.remote.${name}`);
+
+          if (acknowledged && !cluster) {
+            return null;
```

Comment:
A short comment here explaining why we are not returning a `KibanaResponse` for the happy path would be helpful 🙂

Reply:
Added a comment. The response is being returned on [L104](https://github.com/elastic/kibana/pull/56781/files/14cbcb6cb1825fc89c6907a0d34b3231956158c6#diff-b0d61bb65a509d3d026dec9198a72ab0R104).